### PR TITLE
initial build and test workflow

### PR DESCRIPTION
**Pull request by jeffdiclemente**
moving from travis ci to github workflows and updating minimum compiler versions for linux

Signed-off-by: The MathWorks, Inc. <jdicleme@mathworks.com>

### CODE REVIEW

File: .github/workflows/buildandtest.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+         submodules: true
+
+    - name: Install valgrind
+      run: sudo apt-get install -y valgrind
+      if: ${{matrix.os == 'ubuntu-18.04'}}
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DUS_BUILD_TESTING:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=$${{matrix.sharedlib}} -DUS_BUILD_EXAMPLES:BOOL=ON -DUS_ENABLE_THREADING_SUPPORT:BOOL=${{matrix.threading}}
```
Comment (achristoforides) [inline .github/workflows/buildandtest.yml:41]: Why are there two "$" symbols in front of "{{matrix.sharedlib}}"? Earlier in the line, there is only one "$" in front of "{{env.BUILD_TYPE}}".

Comment (jeffdiclemente) [reply .github/workflows/buildandtest.yml:41]: $$ was a typo. it should be $

File: .github/workflows/buildandtest.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+name: BuildAndTest
+
+on:
+  push:
+    branches: [ development ]
+  pull_request:
+    branches: [ development ]
```
Comment (achristoforides) [inline .github/workflows/buildandtest.yml:7]: Does this mean that CI only runs on PRs and subsequent commits in that PR and not on branches without a PR?

Comment (jeffdiclemente) [reply .github/workflows/buildandtest.yml:7]: yes, that's my understanding

**Review by achristoforides — APPROVED** [review]

**Review by pelliott-mathworks — COMMENTED** [review]
Approved.

**Pull request merged**